compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5856,31 +5856,92 @@ private void handleSafeNavigation(BLangAccessExpression accessExpr, BType type,
 
         BLangMatch matchStmt = ASTBuilderUtil.createMatchStatement(accessExpr.pos, accessExpr.expr, new ArrayList<>());
 
+        boolean allRecords = false;
+        BUnionType errorLiftedType = BUnionType.create(null, new LinkedHashSet<>());
+        if (accessExpr.expr.type.tag == TypeTags.UNION) {
+            LinkedHashSet<BType> memTypes = new LinkedHashSet<>(((BUnionType) accessExpr.expr.type).getMemberTypes());
+            errorLiftedType = BUnionType.create(null, memTypes);
+            allRecords = checkForRecords(memTypes);
```

Comment:
IMO the method name does not say much about what the method actually does, and the variable name kinda implies the check returns if all the members are records, which from the impl. doesn't seem to be the case. Can we rename?